### PR TITLE
Rearrange only selected art boards

### PR DESCRIPTION
**Pull request by tpoulos**
Fix issue #3

### CODE REVIEW

**Comment by tpoulos** [comment]
Hey @romannurik, have you had a chance to look at this yet?

**Comment by romannurik** [comment]
Hey there, just tried it out, thanks for doing this. A couple comments, very open to feedback:

- The rearrange command is often used with the renumber command. It's a bit odd if the behavior of rearrange works different than renumber. Then again, I'm not sure what the expected behavior should be for renumbering only selected artboards.

- Right now if you have a non-artboard selected, it still thinks you want to rearrange only selected artboards. A better behavior might be to either (a) find the artboards containing all selected layers and rearrange those... there's code for that in `select_artboards....cocoascript` in this project. (b) just rearrange all artboards if only non-artboards are selected or (c) just rearrange all artboards if at least one non-artboard layer is selected

**Comment by tpoulos** [comment]
Sweet! These are really good comments. 

Here's what I'm thinking: 

It makes sense to only renumber selected art boards. I'll go ahead and make that change.

Do you have any preference on the three behaviors you suggested? My thinking was this:

There are two things that can go wrong here: 
- Selecting layers that they didn't intend to
- Selecting a layer when they intended to select an artboard

Under option A, there's a risk that a user ends up rearranging an artboard when they select a layer they didn't intend to. 

Under option C, there's a chance that they end up rearranging all of the artboards when they selected a layer and intended to select an artboard.

However, under option B,  if they selected a layer and didn't intend to, nothing bad will happen, and if they just select a layer when they intended to select an artboard, then only that artboard won't get moved, allowing them to identify the problem really easily.

As I was working on this PR, my intention was option B. When I try to do it with my test file now, though, it's not working correctly. Do you know of any way to set up an automated testing suite with sketch plugins?

**Comment by tpoulos** [comment]
@romannurik how is this looking now? 

I've set it up to renumber only the selected artboards, and went with option B for the selected a non-artboard behavior.

**Comment by tpoulos** [comment]
Hey @romannurik, have you had a chance to look at this yet?

**Comment by romannurik** [comment]
Sorry haven't had a chance yet. Will take a look as soon as I have a moment to run through open GitHub reqs

**Comment by romannurik** [comment]
Hey there! Thanks again for taking this on. I took it for a spin, and found a few issues:

 - Re-arranging moves selected artboards to the top of your layer list. It should probably only re-arrange the selected artboards in the layer list, and not move them to the top
 - After re-numbering selected artboards, other artboards on your canvas lose their border and title... not sure what's going on there?

Here's a screencap:

screenflow

File: ArtboardTricks.sketchplugin/Contents/Sketch/util.js [hunk 1]
```diff
@@ -24,3 +24,28 @@ function setSelection(context, layers) {
     [l select:true byExpandingSelection:true];
   });
 }
+
+// Returns a list of artboards
+// If at least one artboard is selected, return only the selected artboards
+// If no artboards are selected, then return all of the artboards on the page
+function getArtboards(context, page) {
```
Comment (romannurik) [inline ArtboardTricks.sketchplugin/Contents/Sketch/util.js:31]: this probably needs a more descriptive title, e.g. `getSelectedOrAllArtboards`

**Pull request closed without merging**